### PR TITLE
OOP school library: decorate a class

### PR DESCRIPTION
**Pull request by oliverSCZ**
I did the following :

- Created a class Nameable.
- Implemented a method "correct_name"
- Turned Person class to Nameable class
- Created Decorator, CapitalizeDecorator and TrimmerDecorator classes

### CODE REVIEW

File: nameable.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    raise NotImplementedError
+  end
+end
+
+class Decorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+end
```
Comment (mmsesay) [inline nameable.rb]: - [x] As per the requirements, I believe you forgot to Implement the correct_name method that returns the result of the correct_name method of the @namable.

See example:
```ruby
def correct_name
   @nameable.correct_name
end
```

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  def of_age?
+    @age >= 18
+  end
+end
+
+person = Person.new(22, 'maximilianus')
+person.correct_name
+capitalized_person = CapitalizeDecorator.new(person)
+capitalized_person.correct_name
+capitalized_trimmed_person = TrimmerDecorator.new(capitalizedPerson)
```
Comment (mmsesay) [inline person.rb]: - [x] I think you have a typo here that crashed the app when I tried to test it out. IThe correct variable name should be `capitalized_person`.

See error screenshot:
a1

**Review by mmsesay — CHANGES_REQUESTED** [review]
Hi @oliverSCZ, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mmsesay — CHANGES_REQUESTED** [review]
:+1:

File: nameable.rb (hunk 1 above)
Comment (oliverSCZ) [reply nameable.rb]: Yes, thank you. I forgot put it.

File: person.rb (hunk 2 above)
Comment (oliverSCZ) [reply person.rb]: Yes, sorry. I changed to snake_case and I forgot to change that variable.

**Review by er-contreras — APPROVED** [review]
Hi @oliverSCZ, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

- Well done!

Felix

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**